src/Interpreters/BloomFilterHash.h

```diff
@@ -196,18 +196,17 @@ struct BloomFilterHash
             const ColumnString::Chars & data = index_column->getChars();
             const ColumnString::Offsets & offsets = index_column->getOffsets();
 
-            ColumnString::Offset current_offset = pos;
```

Comment:
We just need to change this to `ColumnString::Offset current_offset = offsets[pos - 1];` to fix this bug.

: (
But I don't know why it affects all the data (in fact, it should only affect the first row of data)

Reply:
>We just need change this to ColumnString::Offset current_offset = offsets[pos - 1]; to fix this bug.

Yes, but to avoid double assignment, I just moved into the loop

>But I don't know why it affects all the data (in fact, it should only affect the first row of data)

You are right, and to be more precise, it affects the first row per index_granularity , but not the first row in block (detailed description reflects that).